action/pipeline/validate.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package pipeline
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/go-vela/compiler/compiler"
+)
+
+// Validate verifies the configuration provided.
+func (c *Config) Validate() error {
+	// check if pipeline file is set
+	if len(c.File) == 0 {
+		return fmt.Errorf("no pipeline file provided")
+	}
+
+	return nil
+}
+
+// ValidateFile verifies a pipeline based off the provided configuration.
+func (c *Config) ValidateFile(client compiler.Engine) error {
```

Comment:
This is where we add the new `ValidateFile()` function.

The purpose of this function is to verify a pipeline configuration (`.yml`/`.yaml`) file is valid based off feedback from the compiler.